lambdas/section-manager-lambda/src/utils.ts

```diff
@@ -129,9 +149,47 @@ export const processSqsSectionData = async (
       // continue to next item
     }
   }
+  // Remove "old" SectionItems not present in ML payload AFTER new ones are created
+  if (sectionItemsToRemove.length > 0) {
+    for (const item of sectionItemsToRemove) {
+      try {
+        await removeSectionItem(config.adminApiEndpoint, graphHeaders, {
+          externalId: item.externalId,
+          deactivateReasons: [SectionItemRemovalReason.ML],
+          deactivateSource: CorpusItemSource.ML,
+        });
+      } catch (e) {
+        console.error(`Failed to remove SectionItem ${item.externalId}`, e);
+        Sentry.captureException(e);
+      }
+    }
+  } else {
+    console.log(`No SectionItems to remove for Section ${sectionExternalId}`);
+  }
+
   console.log(`processSqsSectionData result: ${successfulCandidates} succeeded, ${failedCandidates} failed`);
 };
 
+/**
+ * helper function to compute the diff between currently active SectionItems & SectionItems in ML payload.
+ *
+ * @param sqsSectionData
+ * @param currentActiveSectionItems
+ */
+export const computeSectionItemsToRemove = (
+  sqsSectionData: SqsSectionWithSectionItems,
+  currentActiveSectionItems: ActiveSectionItem[]
```

Comment:
if you wanted to do a little less processing in here, i think you could just pass `activeSectionItemsUrlsSet`.